packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -0,0 +1,139 @@
+  isDevHub?: boolean;
+  username: string;
+}
+const readFileAsync = promisify(fs.readFile);
+export class OrgList {
+  public async getAuthInfoObjects() {
+    const authFilesArray = await AuthInfo.listAllAuthFiles().catch(err => null);
+
+    if (authFilesArray === null || authFilesArray.length === 0) {
+      return null;
+    }
+    const authInfoObjects = Promise.all(
+      authFilesArray.map(fileName => {
+        const filePath = path.join(os.homedir(), '.sfdx', fileName);
+        return readFileAsync(filePath, 'utf8')
+          .then(fileData => JSON.parse(fileData))
+          .catch(err => console.log(err));
+      })
+    );
+    return authInfoObjects;
+  }
+
+  public async filterAuthInfo(authInfoObjects: FileInfo[]) {
+    authInfoObjects = authInfoObjects.filter(fileData =>
+      isNullOrUndefined(fileData.scratchAdminUsername)
+    );
+    const authUsernames = authInfoObjects.map(file => file.username);
+    const aliases = await Aliases.create(Aliases.getDefaultOptions());
+    const authList = [];
+    for (const username of authUsernames) {
+      const alias = await aliases.getKeysByValue(username);
+      if (alias.length > 0) {
+        authList.push(alias + ' - ' + username);
+      } else {
+        authList.push(username);
+      }
+    }
+    return authList;
+  }
+}
+
+let statusBarItem: vscode.StatusBarItem;
+
+export async function updateOrgList() {
+  const orgList = new OrgList();
+  const authInfoObjects = await orgList.getAuthInfoObjects();
+  if (isNullOrUndefined(authInfoObjects)) {
+    return null;
+  }
+  const authUsernameList = await orgList.filterAuthInfo(authInfoObjects);
+  return authUsernameList;
+}
+
+export async function setDefaultOrg(): Promise<
+  CancelResponse | ContinueResponse<{}>
+> {
+  let quickPickList = [
+    '$(plus) ' + nls.localize('force_auth_web_login_authorize_org_text'),
+    '$(plus) ' + nls.localize('force_org_create_default_scratch_org_text')
+  ];
+  const orgList = await updateOrgList();
+  if (!isNullOrUndefined(orgList)) {
+    quickPickList = quickPickList.concat(orgList);
+  }
+  const selection = await vscode.window.showQuickPick(quickPickList, {
+    placeHolder: nls.localize('org_select_text')
+  });
+  if (!selection) {
+    return { type: 'CANCEL' };
+  }
+
+  if (
+    selection ===
+    '$(plus) ' + nls.localize('force_auth_web_login_authorize_org_text')
+  ) {
+    vscode.commands.executeCommand('sfdx.force.auth.web.login');
+    return {
+      type: 'CONTINUE',
+      data: {}
+    };
+  } else if (
+    selection ===
+    '$(plus) ' + nls.localize('force_org_create_default_scratch_org_text')
+  ) {
+    vscode.commands.executeCommand('sfdx.force.org.create');
+    return { type: 'CONTINUE', data: {} };
+  } else {
+    const usernameOrAlias = selection.split(' ', 1);
+    vscode.commands.executeCommand('sfdx.force.config.set', usernameOrAlias);
+    return { type: 'CONTINUE', data: {} };
+  }
+}
+
+export async function showOrg() {
```

Comment:
Should probably change this from `showOrg` to `showOrgListQuickPick` or something like that. Right now it can get confused by the method in `scratch-org-decorator.ts` which is also used in `salesforcedx-vscode-core/src/index.ts`